lib/devise/failure_app.rb

```diff
@@ -38,7 +38,7 @@ def respond
       if http_auth?
         http_auth
       elsif warden_options[:recall]
-        recall
+        request_format == :turbo_stream ? redirect : recall
```

Comment:
I am afraid that while redirecting works in the sense that it will re-display the form as expected, it will lose some context about the flash / error message and show the more generic "you need to be signed in" instead of the "invalid email or password". 

Ideally we'd still recall, but change the status, so it has that same behavior.